Bug fix/annotations list

tmarkn opened the pull request:
- added json uploading along side txt
- added alerts for invalid files
- moved deletelist out of tagModel to app.js (wasn't needed)
- redid annotation list (correctly updates with label rename and deletion) (also added colors)
- fixed a bug with sorcery (no matches throws an error)
- fixed a bug with label deletion (deleting a label wouldn't show which new label was selected)

tmarkn committed:
moved deletelist to app.js

wasn't needed in tagmodel

tmarkn committed:
fixed annotation list

adding + deleting works
style update
now with colors

tmarkn committed:
added json upload

+ alert on invalid files / duplicate files
+ sorcery tweeking

keithudock commented:
can you fix the merge conflict?

tmarkn committed:
Merge branch 'staging' into BugFix/AnnotationsList

tmarkn committed:
fixed bug deleting wrong label

keithudock reviewed (dismissed):
noice

NicoG96 commented on static/js/app.js:
```diff
@@ -114,11 +136,12 @@ textArea.on('mouseup', function (e) {
+  deleteList = tagModel.currentDoc.getAnnotationsAtPos(position);
+
+  if (deleteList.length > 0) {
     $('#delete-menu').append('<h6>Delete Annotation:</h6><hr style="margin: 0;">')
```
#delete-menu is duplicated a lot, you should store it in a variable

tmarkn replied:
updated

NicoG96 commented on static/js/app.js:
```diff
@@ -158,7 +180,7 @@ $('#add-label').on('click', function () {
 
 //change the document's label context
-$('#label-list').on('click', '.label', function () {
+$('#label-list').on('mouseup', '.label', function () {
```
I created a variable for this already, you should refactor a little to use it

tmarkn replied:
updated

tmarkn committed:
code cleanup

made variable $('delete-menu') to variable
condensed the on delete-menu clicks to one function
